### DISCUSSION TITLE
Which naming to use for layout tile splits

### DISCUSSION THREAD

**Discussion by hertg** [leftwm/leftwm#1020] [Polls]
The [leftwm-layouts](https://github.com/leftwm/leftwm-layouts) library, which will be integrated soon, has the concept of `LayoutDefinition`. This is a struct that allows customizing layouts while still being quite strict about the core usability of how LeftWM layouts look and behave.

In this library there is the concept of [`Split`](https://docs.rs/leftwm-layouts/latest/leftwm_layouts/geometry/enum.Split.html). An enum that defines how a particular tile is split up into more tiles. It consists of the following options: `Horizontal`, `Vertical`, `Grid`, `Fibonacci`, `Dwindle`.

What naming would you prefer? Specifically concerning the meaning of `Horizontal` and `Vertical`.

## Option A (Current)
The enum describes **the cuts made when splitting up** a tile

> **Note** This felt like a good choice first, but it may lead to mix-up between horizontal and vertical because horizontal means "cutting the tile by horizontal cuts", but the resulting set of tiles has a vertical "orientation". If you want to have a layout with a vertical stack, it may feel strange to define it as `Stack { split: Horizontal }`. However, this naming seems more in line with how I defined the [`Flip`](https://docs.rs/leftwm-layouts/latest/leftwm_layouts/geometry/enum.Flip.html) enum.

```txt
+--------+        +--+--+--+        +---+---+        +---+---+        +---+---+
|        |        |  |  |  |        |   |   |        |   |   |        |   |   |
+--------+        |  |  |  |        |   |   |        |   |   |        |   |   |
|        |        |  |  |  |        |   +---+        |   +-+-+        |   +-+-+
+--------+        |  |  |  |        |   |   |        |   |_| |        |   | |_|
|        |        |  |  |  |        |   |   |        |   | | |        |   | |||
+--------+        +--+--+--+        +---+---+        +---+---+        +---+---+
                            
horizontal         vertical           grid           fibonacci         dwindle
```

## Option B
The enum describes the **orientation of the resulting tiles** after splitting up a larger one. 

> **Note** I think this will lead to less confusion when users define their own layouts, because `Vertical` means you end up with a vertical stack. It could be defined as `Stack { split: Vertical }`. It does seem inconsistent with [`Flip`](https://docs.rs/leftwm-layouts/latest/leftwm_layouts/geometry/enum.Flip.html), but maybe that one needs changing too? I just don't know how to write its documentation if I were to switch it up.

```txt
+--------+        +--+--+--+        +---+---+        +---+---+        +---+---+
|        |        |  |  |  |        |   |   |        |   |   |        |   |   |
+--------+        |  |  |  |        |   |   |        |   |   |        |   |   |
|        |        |  |  |  |        |   +---+        |   +-+-+        |   +-+-+
+--------+        |  |  |  |        |   |   |        |   |_| |        |   | |_|
|        |        |  |  |  |        |   |   |        |   | | |        |   | |||
+--------+        +--+--+--+        +---+---+        +---+---+        +---+---+
                            
 vertical         horizontal          grid           fibonacci         dwindle
```

**Comment by Antikyth** [comment]
I think using just the names `Vertical` and `Horizontal` is always going to lead to confusion.  Perhaps having either the names `VerticalStack` and `HorizontalStack` or the names `VerticalSplit` and `HorizontalSplit` would be better.

If the names were more descriptive like this, I would prefer the stack ones to the split ones.

**Reply by hertg** [reply]
I think I'll probably go with Option A, based on the current poll results. Your comment makes sense, but in the context of the enum this information is already provided `Split::Horizontal`, `Split::Vertical`, so explicitly adding the `Split` suffix would be `Split::HorizontalSplit` which is a bit awkward.

Seeing it like that, I think going with Option A is most sensible. Option B would only really make sense if I were to rename the enum itself to something different than `Split`.

**Comment by Antikyth** [comment]
Also, the link to `Flipped` is dead.

**Reply by hertg** [reply]
Renamed some stuff in the latest release, I updated the links
